### ISSUE TITLE
Batched inference

### ISSUE THREAD

**Issue by simoneVU** [ZhengPeng7/BiRefNet#150]
Hi,

I was wondering whether it is possible to run batched inference with the BiRefNet tiny model. I tried it with HF but it does not seem to provide me any runtime advantage.

This is my current implementation:

```python
def process_batch(birefnet, image_paths):
    original_images = [Image.open(path) for path in image_paths]
    input_images = torch.stack([transform_image(img) for img in original_images]).to('cuda')

    with torch.no_grad():
        preds = birefnet(input_images)[-1].sigmoid().cpu()

    results = []
    for i, pred in enumerate(preds):
        pred_pil = transforms.ToPILImage()(pred.squeeze())
        mask = pred_pil.resize(original_images[i].size)
        segmented_image = original_images[i].copy()
        segmented_image.putalpha(mask)
        results.append((segmented_image, mask))

    return results

def main():
    start_time = time.time()
    image_paths = list(Path(INPUT_FOLDER).glob("*"))
    batch_count = 0
    
    for i in range(0, len(image_paths), BATCH_SIZE):
        batch_count = batch_count + 1
        batch_paths = image_paths[i:i+BATCH_SIZE]
        batch_results = process_batch(birefnet, batch_paths)

        for path, (segmented_image, mask) in zip(batch_paths, batch_results):
            image_name = os.path.splitext(os.path.basename(path))[0]
            output_subfolder = os.path.join(OUTPUT_FOLDER, image_name)
            
            save_image(segmented_image, f"{image_name}_segmented.png", output_subfolder)
            save_image(mask, f"{image_name}_mask.png", output_subfolder)
```
Thanks in advance!

**Comment by ZhengPeng7** [comment]
You're welcome! In my opinion, that's because even the batch size is only 1, the inference may cost near 100% GPU usage due to the high cost of the inference on `1024x1024` sample. So, although the GPU is able to to take a larger batch size, there is no acceleration.
